src/OpenFeature/EventExecutor.cs

```diff
@@ -0,0 +1,294 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using OpenFeature.Constant;
+using OpenFeature.Model;
+
+namespace OpenFeature
+{
+    internal class EventExecutor
+    {
+        private readonly Mutex _mutex = new Mutex();
```

Comment:
I think that a mutex is heavier than this needs, because it is a basic locking situation within 1 app domain. Could we use a `lock` statement with a lock object instead?

Reply:
sure, will look into that